tsdb/exemplar.go

```diff
@@ -69,12 +71,15 @@ type circularBufferEntry struct {
 
 // If we assume the average case 95 bytes per exemplar we can fit 5651272 exemplars in
 // 1GB of extra memory, accounting for the fact that this is heap allocated space.
-func NewCircularExemplarStorage(len int, reg prometheus.Registerer) *CircularExemplarStorage {
+func NewCircularExemplarStorage(len int, reg prometheus.Registerer) (*CircularExemplarStorage, error) {
+	if len < 1 {
+		return nil, errors.Errorf("must initialize exemplar storage with space for at least 1 exemplar, %d is invalid", len)
```

Comment:
Shouldn't 0 be valid?
Can you use fmt.Errorf intead?

Reply:
I don't think we should allow 0. Do we still want to have exemplar storage behind a feature flag? Either way, allowing users to set the # of exemplars stored to 0 is just another way they could shoot themselves in the foot that we could have prevented.